### PR TITLE
Dev to Main

### PR DESCRIPTION
**Pull request by invaderb**
feat: add sagger documentation and decorators
chore: update game-history uri to add versioning
test: add unit tests for ptn service and game search
chore: increase throttling
chore: update deploy actions
chore: add unit test to CI

### CODE REVIEW

**Review by gruppler — APPROVED** [review]

File: src/models/events/events.controller.ts [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
+	})
+	@ApiResponse({
+		status: 404,
+		type: DefaultExceptionDto,
+		description: 'Returns 404 server error',
+	})
+	@ApiResponse({
+		status: 429,
+		type: DefaultExceptionDto,
+		description: 'Returns 429 too many reuests error',
```
Comment (nitzel) [inline src/models/events/events.controller.ts:26]: re**q**uests

File: src/models/games-history/dto/games.dto.ts [hunk 2]
```diff
@@ -1,4 +1,167 @@
...
+		required: false
+	})
+		sort?: string;
+	@ApiProperty({
+		description: 'Game id',
+		required: false
+	})
+		id?: number;
+	@ApiProperty({
+		description: 'PLayer white username',
```
Comment (nitzel) [inline src/models/games-history/dto/games.dto.ts]: `PLayer` -> `Player`

File: src/models/games-history/games.controller.ts [hunk 3]
```diff
@@ -1,25 +1,117 @@
...
+	})
+	@ApiResponse({
+		status: 404,
+		type: DefaultExceptionDto,
+		description: 'Returns 404 server error',
+	})
+	@ApiResponse({
+		status: 429,
+		type: DefaultExceptionDto,
+		description: 'Returns 429 too many reuests error',
```
Comment (nitzel) [inline src/models/games-history/games.controller.ts]: `reuests` -> `requests`

File: src/models/games-history/games.controller.ts [hunk 4]
```diff
@@ -1,25 +1,117 @@
...
```
Comment (nitzel) [inline src/models/games-history/games.controller.ts]: `reuests` -> `requests`

File: src/models/games-history/games.controller.ts [hunk 5]
```diff
@@ -1,25 +1,117 @@
...
```
Comment (nitzel) [inline src/models/games-history/games.controller.ts]: `reuests` -> `requests`

File: src/models/games-history/games.controller.ts [hunk 6]
```diff
@@ -1,25 +1,117 @@
...
```
Comment (nitzel) [inline src/models/games-history/games.controller.ts]: `reuests` -> `requests`

Comment (invaderb) [reply src/models/games-history/games.controller.ts]: ah good catch copy paste error

**Pull request merged**